time stamp
int32 location
bool move
